### PR TITLE
feat(benchmarks): Add basic dart transformer benchmarks.

### PR DESCRIPTION
**Pull request by jakemac53**
Adds simple benchmarks for various transformation phases, as well as hello_world.
Does not integrate these into any benchmark frameworks yet.

### CODE REVIEW

**Comment by jakemac53** [comment]
I think the idea is to just integrate these into golem, but if we want to integrate them elsewhere let me know and point me in the right direction and I can figure that out.

**Comment by yjbanov** [comment]
Reassigning to @kegluneq for review. One comment from me would be that we should move the benchmark code to `test` directory and run a dry run of this benchmark as part of our unit-tests.

File: modules/angular2/benchmark/transform/integration/hello_world.dart [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+''';
+
+const indexCommonContents = '''
+library examples.src.hello_world.index_common;
+
+import "package:angular2/angular2.dart"
+    show bootstrap, Component, Decorator, Template, NgElement;
+import "package:angular2/di.dart"
+    show Injectable; // Angular 2.0 supports 3 basic types of directives:
```
Comment (kegluneq) [inline modules/angular2/benchmark/transform/integration/hello_world.dart]: Remove these comments -- some are already out of date

Comment (jakemac53) [inline modules/angular2/benchmark/transform/integration/hello_world.dart]: Good point no need for any of them for the benchmark

**Comment by kegluneq** [comment]
I wouldn't be opposed to moving these into `test/`. Another potential benefit of that would be that we could use the existing files rather than having the code inlined -- some of the `.ng_deps.dart` formatting is still changing, so we'd save ourselves some potential maintenance.

I don't feel strongly about it though, @jakemac53 , what do you think?

File: modules/angular2/benchmark/transform/integration/hello_world.dart (hunk 1 above)
Comment (jakemac53) [inline modules/angular2/benchmark/transform/integration/hello_world.dart]: done

**Comment by jakemac53** [comment]
I do like the idea of moving the files into test just so that we can make sure they continue to execute as part of the unit tests.

I don't know that we really want to use the actual files on disk though since we don't want changes to those files to affect our benchmark (the applications we are benchmarking should be as written in stone as possible, and the tests are somewhat likely to change).

**Comment by jakemac53** [comment]
updated, ptal @kegluneq

**Comment by jakemac53** [comment]
updated to get the benchmarks running with the unit tests, once that gets the green light I will submit

**Pull request merged**